sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -422,6 +426,15 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
             }
             metrics.foreach(m => checkMetric(m, m))
 
+          // see Analyzer.ResolveMelt
+          case m: Melt if m.childrenResolved && m.ids.forall(_.resolved) && m.values.isEmpty =>
```

Comment:
I'm not sure if we should do this check here. It's possible that we have not infer the value columns yet when hitting this rule, and we should wait for other rules (`ResolveMelt`) to infer value columns first.

I think we should move this check to `CheckAnalysis`